iothub_client/tests/common_longhaul/iothub_client_statistics.c

```diff
@@ -1181,6 +1196,16 @@ int iothub_client_statistics_get_device_method_summary(IOTHUB_CLIENT_STATISTICS_
 
                     summary->methods_received = summary->methods_received + 1;
                 }
+                else if (device_method_info->method_result == IOTHUB_DEVICE_METHOD_ERROR)
+                {
+                    // remove the item if we failed to send the C2D message
+                    summary->methods_received--;
```

Comment:
Analogous feedback for C2D case...

I don't understand how this fits together from my code review.  In `iothub_client_statistics_get_device_method_summary` we're marching through the list and in the `if (device_method_info->time_received != INDEFINITE_TIME)` block we increment the `methods_received`.  Makes sense.

But if we have an error, we end up decrementing.  Which we shouldn't need to do since we never incremented in 1st place?  So like if all the methods failed, we could end up with a negative number in methods_received?

I also wonder if the statistics of the # of errors is interesting in its own right?

Maybe(??) something like
```c
if (device_method_info->time_invoked != INDEFINITE_TIME)
{ 
   ...
   if (device_method_info->method_result == IOTHUB_DEVICE_METHOD_OK)
   {
      summary->methods_received++;
   }
   else
   {
    summary->methods_failed++; // new field
   }
}    
```

Reply:
oops. this should have been methods_invoked